xmr-btc/tests/harness/storage.rs

```diff
@@ -0,0 +1,157 @@
+use anyhow::{anyhow, Context, Result};
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use std::path::Path;
+
+pub struct Database {
+    db: sled::Db,
+}
+
+impl Database {
+    const LAST_STATE_KEY: &'static str = "latest_state";
```

Comment:
If we use several keys we will have to implement logic to know which key corresponds to the latest state. It is not difficult to do but it does increase the number of moving parts in the codebase.

We can achieve type safety by saving the parent state enum instead of saving the enum variants. To resume protocol execution I have to convert the enum variant to the parent enums as the protocol executor loop matches to determine the variant anyway.

Would this be sufficient?

Reply:
Alright going to merge this as is and make the changes we discussed in a follow up PR once the usage is more clear